RegionMaskingFilter not a method

### Server Implementation

Paper

### Server Version

1.20.4

### Describe the bug

I tried using the RegionMaskingFilter on an region but it gave not a method.

Stacktrace:

```[23:04:28 ERROR]: null
org.bukkit.command.CommandException: Unhandled exception executing command 'startlevel' in plugin TheBackrooms v1.0
        at org.bukkit.command.PluginCommand.execute(PluginCommand.java:47) ~[paper-api-1.20.4-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.command.SimpleCommandMap.dispatch(SimpleCommandMap.java:155) ~[paper-api-1.20.4-R0.1-SNAPSHOT.jar:?]
        at org.bukkit.craftbukkit.v1_20_R3.CraftServer.dispatchCommand(CraftServer.java:999) ~[paper-1.20.4.jar:git-Paper-499]
        at org.bukkit.craftbukkit.v1_20_R3.command.BukkitCommandWrapper.run(BukkitCommandWrapper.java:64) ~[paper-1.20.4.jar:git-Paper-499]
        at com.mojang.brigadier.context.ContextChain.runExecutable(ContextChain.java:73) ~[brigadier-1.2.9.jar:?]
        at net.minecraft.commands.execution.tasks.ExecuteCommand.a(ExecuteCommand.java:31) ~[paper-1.20.4.jar:git-Paper-499]
        at net.minecraft.commands.execution.tasks.ExecuteCommand.execute(ExecuteCommand.java:19) ~[paper-1.20.4.jar:git-Paper-499]
        at net.minecraft.commands.execution.UnboundEntryAction.a(UnboundEntryAction.java:8) ~[paper-1.20.4.jar:git-Paper-499]
        at net.minecraft.commands.execution.CommandQueueEntry.a(CommandQueueEntry.java:5) ~[paper-1.20.4.jar:git-Paper-499]
        at net.minecraft.commands.execution.ExecutionContext.a(ExecutionContext.java:103) ~[paper-1.20.4.jar:git-Paper-499]
        at net.minecraft.commands.CommandDispatcher.a(CommandDispatcher.java:434) ~[paper-1.20.4.jar:git-Paper-499]
        at net.minecraft.commands.CommandDispatcher.performCommand(CommandDispatcher.java:336) ~[paper-1.20.4.jar:git-Paper-499]
        at net.minecraft.commands.CommandDispatcher.a(CommandDispatcher.java:323) ~[paper-1.20.4.jar:git-Paper-499]
        at net.minecraft.server.network.PlayerConnection.a(PlayerConnection.java:2230) ~[paper-1.20.4.jar:git-Paper-499]
        at net.minecraft.server.network.PlayerConnection.lambda$handleChatCommand$14(PlayerConnection.java:2190) ~[paper-1.20.4.jar:git-Paper-499]
        at net.minecraft.util.thread.IAsyncTaskHandler.b(IAsyncTaskHandler.java:59) ~[paper-1.20.4.jar:git-Paper-499]
        at java.util.concurrent.CompletableFuture$AsyncSupply.run(CompletableFuture.java:1768) ~[?:?]
        at net.minecraft.server.TickTask.run(TickTask.java:18) ~[paper-1.20.4.jar:git-Paper-499]
        at net.minecraft.util.thread.IAsyncTaskHandler.d(IAsyncTaskHandler.java:149) ~[paper-1.20.4.jar:git-Paper-499]
        at net.minecraft.util.thread.IAsyncTaskHandlerReentrant.d(IAsyncTaskHandlerReentrant.java:24) ~[paper-1.20.4.jar:git-Paper-499]
        at net.minecraft.server.MinecraftServer.b(MinecraftServer.java:1465) ~[paper-1.20.4.jar:git-Paper-499]
        at net.minecraft.server.MinecraftServer.d(MinecraftServer.java:194) ~[paper-1.20.4.jar:git-Paper-499]
        at net.minecraft.util.thread.IAsyncTaskHandler.x(IAsyncTaskHandler.java:123) ~[paper-1.20.4.jar:git-Paper-499]
        at net.minecraft.server.MinecraftServer.bl(MinecraftServer.java:1442) ~[paper-1.20.4.jar:git-Paper-499]
        at net.minecraft.server.MinecraftServer.x(MinecraftServer.java:1365) ~[paper-1.20.4.jar:git-Paper-499]
        at net.minecraft.util.thread.IAsyncTaskHandler.c(IAsyncTaskHandler.java:133) ~[paper-1.20.4.jar:git-Paper-499]
        at net.minecraft.server.MinecraftServer.w_(MinecraftServer.java:1343) ~[paper-1.20.4.jar:git-Paper-499]
        at net.minecraft.server.MinecraftServer.w(MinecraftServer.java:1232) ~[paper-1.20.4.jar:git-Paper-499]
        at net.minecraft.server.MinecraftServer.lambda$spin$0(MinecraftServer.java:319) ~[paper-1.20.4.jar:git-Paper-499]
        at java.lang.Thread.run(Thread.java:1583) ~[?:?]
Caused by: java.lang.NoSuchMethodError: 'void com.sk89q.worldedit.function.RegionMaskingFilter.<init>(com.sk89q.worldedit.function.mask.Mask, com.sk89q.worldedit.function.RegionFunction)'
        at collinvht.backrooms.module.main.module.levelsystem.object.levels.level37.Level37.loadExtra(Level37.java:89) ~[Backrooms-1.0.jar:?]
```



### To Reproduce

1. Write a piece of code for an custom selection, as seen below
```
Region region = new CuboidRegion(Utils.getAdaptedWorld(world), BlockVector3.at(-37, -64, 128), BlockVector3.at(228, 319, -139));
                Region region2 = new CuboidRegion(Utils.getAdaptedWorld(world), BlockVector3.at(128, -64, 62), BlockVector3.at(169, 319, 39));
for (UUID uuid : session.getParty().getPartyList()) {
    Player player1 = Bukkit.getPlayer(uuid);
    if (player1 != null && player1.isOnline()) player = player1;
}
if (player == null) return;
EditSession editSession = Utils.getWorldEdit().createEditSession(player);
```
2. Then add the following line
```
replace = new RegionMaskingFilter(mask, replace);
````
3. Let the code run (Thru a command or any event)
4. See the stacktrace from the previous step

### Expected behaviour

I'd expect it to mask out the other region

### Screenshots / Videos

_No response_

### Error log (if applicable)

https://gist.github.com/Collinvh/69f7790f45e2da48a8a1b476b99d1fc5

### Fawe Debugpaste

https://athion.net/ISPaster/paste/view/009e38caaadf47d4b4c4cfa41f180816

### Fawe Version

2.12.2-SNAPSHOT-973;308e909

### Checklist

- [X] I have included a Fawe debugpaste.
- [X] I am using the newest build from https://ci.athion.net/job/FastAsyncWorldEdit/ and the issue still persists.

### Anything else?

_No response_

--- a/worldedit-core/src/main/java/com/sk89q/worldedit/command/BiomeCommands.java
+++ b/worldedit-core/src/main/java/com/sk89q/worldedit/command/BiomeCommands.java
@@ -206,7 +206,7 @@ public class BiomeCommands {
 
         RegionFunction replace = new BiomeReplace(editSession, target);
         if (mask != null) {
-            replace = new RegionMaskingFilter(editSession, mask, replace);
+            replace = new RegionMaskingFilter(mask, replace);
         }
         RegionVisitor visitor = new RegionVisitor(region, replace);
         Operations.completeLegacy(visitor);

--- a/worldedit-core/src/main/java/com/sk89q/worldedit/extent/Extent.java
+++ b/worldedit-core/src/main/java/com/sk89q/worldedit/extent/Extent.java
@@ -818,7 +818,7 @@ public interface Extent extends InputExtent, OutputExtent {
         checkNotNull(pattern);
 
         BlockReplace replace = new BlockReplace(this, pattern);
-        RegionMaskingFilter filter = new RegionMaskingFilter(this, mask, replace);
+        RegionMaskingFilter filter = new RegionMaskingFilter(mask, replace);
         //FAWE start > add extent to RegionVisitor to allow chunk preloading
         RegionVisitor visitor = new RegionVisitor(region, filter, this);
         //FAWE end

--- a/worldedit-core/src/main/java/com/sk89q/worldedit/function/RegionMaskingFilter.java
+++ b/worldedit-core/src/main/java/com/sk89q/worldedit/function/RegionMaskingFilter.java
@@ -35,9 +35,19 @@ public class RegionMaskingFilter implements RegionFunction {
 
     private final RegionFunction function;
     private final Mask mask;
-    //FAWE start
-    private final Extent extent;
-    //FAWE end
+
+    /**
+     * Create a new masking filter.
+     *
+     * @param mask the mask
+     * @param function the function
+     */
+    public RegionMaskingFilter(Mask mask, RegionFunction function) {
+        checkNotNull(function);
+        checkNotNull(mask);
+        this.mask = mask;
+        this.function = function;
+    }
 
     /**
      * Create a new masking filter.
@@ -46,16 +56,11 @@ public class RegionMaskingFilter implements RegionFunction {
      * @param function the function
      */
     //FAWE start - Extent
-    public RegionMaskingFilter(Extent extent, Mask mask, RegionFunction function) {
-        checkNotNull(function);
-        checkNotNull(mask);
-        //FAWE start
-        checkNotNull(extent);
-        this.extent = extent;
-        //FAWE end
-        this.mask = mask;
-        this.function = function;
+    @Deprecated(since = "TODO")
+    public RegionMaskingFilter(@SuppressWarnings("unused") Extent extent, Mask mask, RegionFunction function) {
+        this(mask, function);
     }
+    //FAWE end - Extent
 
     @Override
     public boolean apply(BlockVector3 position) throws WorldEditException {

--- a/worldedit-core/src/main/java/com/sk89q/worldedit/function/operation/ForwardExtentCopy.java
+++ b/worldedit-core/src/main/java/com/sk89q/worldedit/function/operation/ForwardExtentCopy.java
@@ -330,7 +330,7 @@ public class ForwardExtentCopy implements Operation {
                 }
                 if (sourceMask != Masks.alwaysTrue()) {
                     new MaskTraverser(sourceMask).reset(transExt);
-                    copy = new RegionMaskingFilter(source, sourceMask, copy);
+                    copy = new RegionMaskingFilter(sourceMask, copy);
                 }
                 if (copyingBiomes && (source.isWorld() || region instanceof FlatRegion)) {
                     copy = CombinedRegionFunction.combine(copy, new BiomeCopy(source, finalDest));
@@ -394,7 +394,7 @@ public class ForwardExtentCopy implements Operation {
                 if (maskFunc != null) {
                     copy = new RegionMaskTestFunction(sourceMask, copy, maskFunc);
                 } else {
-                    copy = new RegionMaskingFilter(source, sourceMask, copy);
+                    copy = new RegionMaskingFilter(sourceMask, copy);
                 }
             }
             if (copyingBiomes && (source.isWorld() || region instanceof FlatRegion)) {

